File: dotnet/src/App/Api/Models/Author.cs
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore.Metadata;
using System.ComponentModel.DataAnnotations;

namespace App.Api.Models {

    public static class AuthorMeta{

        public static void OnModelCreating(ModelBuilder modelBuilder){

            modelBuilder.Entity<Author>(entity => {
                entity.HasMany(m => m.Posts)
                    .WithOne(m => m.Author)
                    .HasForeignKey(m => m.AuthorId)
                    .HasPrincipalKey(m => m.Id)
                    .OnDelete(DeleteBehavior.Restrict);
            });

        }

    }
    
    public class Author
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public List<Post> Posts { get; set; }
        private bool _showPosts = false;
        public void ShowPosts() {
            _showPosts = true;
        }
        public bool ShouldSerializePosts() => _showPosts;

    }
    
}

File: ui/src/models/Author.ts
import Post from './Post'

export default class Author {

    public static convertAll(models: object[]): Author[] {
        let converted = new Array<Author>()
        for (let model of models) {
            converted.push(new Author(model))
        }
        return converted
    }

    constructor(model?: object){
        if (model){
            Object.assign(this, model)
            if (this.posts){
                this.posts = Post.convertAll(this.posts)
            }
        }
    }

    public id: number;

    public name: string;

    public posts: Post[];

}